extension/awsmiddleware/middleware_test.go

```diff
@@ -161,27 +169,39 @@ func TestAppendOrder(t *testing.T) {
 	}
 	for name, testCase := range testCases {
 		t.Run(name, func(t *testing.T) {
-			middleware := &testMiddlewareExtension{}
 			recorder := &recordOrder{}
+			var requestHandlers []RequestHandler
 			for _, handler := range testCase.requestHandlers {
-				handler.handleRequest = recorder.handle(handler.id)
-				middleware.requestHandlers = append(middleware.requestHandlers, handler)
+				handler.handleRequest = recorder.Handle(handler.id)
+				requestHandlers = append(requestHandlers, handler)
 			}
+			handler := new(MockHandler)
+			handler.On("ID").Return("mock")
+			handler.On("Position").Return(After)
+			handler.On("HandleRequest", mock.Anything, mock.Anything)
+			handler.On("HandleResponse", mock.Anything, mock.Anything)
+			requestHandlers = append(requestHandlers, handler)
+			middleware := new(MockMiddlewareExtension)
+			middleware.On("Handlers").Return(
+				requestHandlers,
+				[]ResponseHandler{handler},
+			)
+			c := NewConfigurer(middleware)
```

Comment:
I am wondering if we can make this easier to use.
Currently the user has to create the `handlers`, then create the `middleware`, then create the `Configurer`.

Could the `NewConfigurer` Function just take in the handlers?

Does this support adding a single request handler, or single response handler or do they always have to be added in pairs?

Reply:
I suppose it could. `middleware` is just a wrapper for the handlers. They don't have to be in pairs. Like the user agent one is only a single request handler.